File: python/prime-factors/prime_factors.py
def prime_factors(number):
    factors = []
    current_factor = 2
    current_number = number
    while current_number > 1:
        while _is_divisible(current_number, current_factor):
            factors.append(current_factor)
            current_number /= current_factor
        current_factor += 1 
    return factors

def _is_divisible(number, divisor):
    return number % divisor == 0

File: javascript/prime-factors/prime-factors.js
(function() {
  'use strict';

  var PrimeFactors = {
    for: function(number) {
      
      var currentFactor = 2;
      var factors = [];
      var currentNumber = number;

      while(currentNumber > 1) {
        while(currentNumber % currentFactor === 0) {
          factors.push(currentFactor)
          currentNumber /= currentFactor;
        }
        currentFactor += 1;
      }
      return factors;
    }
  }


  module.exports = PrimeFactors;

})();